testing/end-to-end/virtualisation-config.nix

```diff
@@ -0,0 +1,35 @@
+{ overlayPath, ... }:
+{
+    config = {
+        # Kinda abusing the NixOS testing infra here, because
+        # there is no other interface for creating test VMs/nodes.
+        #
+        # Instead of specifying/building a NixOS system, here we
+        # pass an already built disk image, so the options below are mainly
+        # for _preventing_ qemu-vm.nix from passing any unnecssary flags to
+        # QEMU.
+        #
+        # Due to this, test driver features requiring
+        # `virtualisation.sharedDirectories` (e.g. `copy_from_vm`) are not
+        # functional.
+        virtualisation.mountHostNixStore = false;
+        virtualisation.useHostCerts = false;
+        virtualisation.directBoot.enable = false;
+        virtualisation.useEFIBoot = true;
+        virtualisation.useBootLoader = false;
+        virtualisation.diskImage = null;
+
+        # good when debugging in interactive mode
+        virtualisation.graphics = true;
+
+        # give it a bit more resources
+        virtualisation.memorySize = 2048;
+        virtualisation.cores = 2;
+
+        virtualisation.qemu.options = [
+            "-enable-kvm"
+            # HACK, see testScript
```

Comment:
Did not manage it in 1 line: ad21888f3f85bb, not sure it got clearer though.

Basically, it is a hack because it produces a VM start script that does not work out of the box, because the overlay path is created "out of band" in the test script. Regular VMs built using the `qemu-vm.nix` tooling can be run directly. I looked into ways to somehow "hook into" the `startVM` script, but it doesn't seem to be possible.

Reply:
Definitely clearer :+1: